include/reactor-cpp/assert.hh

```diff
@@ -8,22 +8,28 @@
 
 #pragma once
 
-// A custom definition of assert that mitigates unused variable warnings
-// when assertions are disabled
-#ifdef NDEBUG
-#define ASSERT(x)    \
-  do {               \
-    (void)sizeof(x); \
-  } while (0)
+#ifdef REACTOR_CPP_VALIDATE
+constexpr bool kDoRuntimeValidation = true;
```

Comment:
What does the k prefix stand for? Also, the snake-case is throughout reactor-cpp to name variables and functions.

Reply:
This comes from here: [https://google.github.io/styleguide/cppguide.html#Constant_Names](https://google.github.io/styleguide/cppguide.html#Constant_Names). 

Because this project uses the `Chromium` style which is a specialization of the google cpp style guide. 

But we maybe want to discuss how appropriate it is to use `Chromium` style guide because it appears that this project is more in line with the `boost` style. e.g. Chromium wants CamelCase for function but we use sneak-case everywhere.